Brad/fix 8103 usability and errors ingestion notebook

Brad-Edwards opened the pull request:
# Description

Developer Experience/Missing Steps
1. Add install dependencies
2. Add instructions to get Pinecone and OpenAI keys
3. Add environment variable loading that works with hosted notebooks
4. Add info/code to for deleting indexes on subsequent runs (for Pinecone Starter accounts)

Errors
1. Replaced PyMuPDF from Llama Hub with vanilla PyMuPDF package to avoid errors from fitz, frontend and Starlette packages

Fixes #8103 

## Type of Change

Please delete options that are not relevant.

- [X] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [X] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Added new unit/integration tests
- [ ] Added new notebook (that tests end-to-end)
- [X] I stared at the code and made sure it makes sense
- [X] Notebook runs as expected

# Suggested Checklist:

- [X] I have performed a self-review of my own code
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] I have made corresponding changes to the documentation
- [X] My changes generate no new warnings
- [X] I have added tests that prove my fix is effective or that my feature works
- [X] New and existing unit tests pass locally with my changes
- [X] I ran `make format; make lint` to appease the lint gods

logan-markewich reviewed (approved):
Thanks for doing this!

A few quick comments
- You specify an `index_name` but then don't use it in `pinecone.Index()`
- curious -- why switch the llamahub loader for fitz? Is fitz something the user should install?

Brad-Edwards commented:
Thanks for catching that. Fixed the `index_name`.

I switched from the llama_hub loader because it was the cause of the errors I noted in the bug report.

To use `from llama_hub.file.pymu_pdf.base import PyMuPDFReader` in the original code, we have to install the llama-hub package. That's fine, but then when we run the code:

- First it throws an error that the `fitz` package is not installed. 
- When `fitz` is installed and we run the code again, we get the error about `frontend` not being installed.
- When `frontend` is installed, `starlette` throws errors that it cannot find the `static/` folder

So the original code seems to be using `fitz`, but pulling in a bunch of other dependencies. By installing `PyMuPDF` at the beginning, we `fitz` and we can just use it directly.

It does remove one reference to our ecosystem, but it's a very minor one that doesn't relate to the core of the tutorial. So I figured it was a reasonable tradeoff.

logan-markewich commented:
Makes sense, thanks for the explanation!
